sql/moz-fx-data-shared-prod/subscription_platform_derived/stripe_logical_subscriptions_history_v1/query.sql

```diff
@@ -0,0 +1,209 @@
+    IF(paypal_subscriptions.subscription_id IS NOT NULL, 'PayPal', 'Stripe') AS payment_provider,
+    history.subscription.id AS provider_subscription_id,
+    subscription_item.id AS provider_subscription_item_id,
+    history.subscription.created AS provider_subscription_created_at,
+    history.subscription.customer_id AS provider_customer_id,
+    history.customer.metadata.userid AS mozilla_account_id,
+    history.customer.metadata.userid_sha256 AS mozilla_account_id_sha256,
+    CASE
+      -- Use the same address hierarchy as Stripe Tax after we enabled Stripe Tax (FXA-5457).
+      -- https://stripe.com/docs/tax/customer-locations#address-hierarchy
+      WHEN DATE(history.valid_to) >= '2022-12-01'
+        AND (
+          DATE(history.subscription.ended_at) >= '2022-12-01'
+          OR history.subscription.ended_at IS NULL
+        )
+        THEN COALESCE(
+            NULLIF(history.customer.shipping.address.country, ''),
+            NULLIF(history.customer.address.country, ''),
+            charge_summaries.latest_card_country
+          )
+      -- SubPlat copies the PayPal billing agreement country to the customer's address.
+      WHEN paypal_subscriptions.subscription_id IS NOT NULL
+        THEN NULLIF(history.customer.address.country, '')
+      ELSE charge_summaries.latest_card_country
+    END AS country_code,
+    plan_services.services,
+    subscription_item.plan.product.id AS provider_product_id,
+    subscription_item.plan.product.name AS product_name,
+    subscription_item.plan.id AS provider_plan_id,
+    subscription_item.plan.`interval` AS plan_interval_type,
+    subscription_item.plan.interval_count AS plan_interval_count,
+    UPPER(subscription_item.plan.currency) AS plan_currency,
+    (CAST(subscription_item.plan.amount AS DECIMAL) / 100) AS plan_amount,
+    IF(ARRAY_LENGTH(plan_services.services) > 1, TRUE, FALSE) AS is_bundle,
+    IF(
+      history.subscription.status = 'trialing'
+      OR (
+        history.subscription.ended_at
+        BETWEEN history.subscription.trial_start
+        AND history.subscription.trial_end
+      ),
+      TRUE,
+      FALSE
+    ) AS is_trial,
+    history.subscription_is_active AS is_active,
+    history.subscription.status AS provider_status,
+    history.subscription_first_active_at AS started_at,
+    history.subscription.ended_at,
+    -- TODO: ended_reason
+    IF(
+      history.subscription.ended_at IS NULL,
+      history.subscription.current_period_start,
+      NULL
+    ) AS current_period_started_at,
+    IF(
+      history.subscription.ended_at IS NULL,
+      history.subscription.current_period_end,
+      NULL
+    ) AS current_period_ends_at,
+    history.subscription.cancel_at_period_end IS NOT TRUE AS auto_renew,
+    IF(
+      history.subscription.cancel_at_period_end,
+      history.subscription.canceled_at,
+      NULL
+    ) AS auto_renew_disabled_at,
+    -- TODO: promotion_codes
+    -- TODO: promotion_discounts_amount
```

Comment:
should these todos block merging? also a few lines above for `ended_reason`

Reply:
No.  In this PR I've tried to limit the scope to just what's needed for the SubPlat metrics, since some SubPlat metrics work is currently blocked waiting for this.  I'm leaving those TODOs as reminders to myself to come back and implement the additional stuff.